module4/lessons/Week1/JavaScript.md

```diff
@@ -0,0 +1,387 @@
+```javascript
+function myCoolFunction() {
+  let myFirstStatement = "Assigning a string to a variable called 'myFirstStatement'";
+  alert('This is my second statement!');
+  console.log(myFirstStatement);
+}
+// because definition of method will not change
+const myFirstStatement = () => {
+  let myFirstStatement = "Assigning a string to a variable called 'myFirstStatement'";
+  alert('This is my second statement!');
+  console.log(myFirstStatement);
+}
+```
+
+- You declare a function using the keyword `function`.
+- You can name the function anything you want, but it must include a set of parentheses after the name, which can be empty or accept parameters.
+- You must follow the name of the function with a set of curly braces, which act as bookends to hold the set of statements you want the function to run when it is called.
+- Calling the function is quite easy. Simply type the function name with it's parenthesis and any associated parameters.
+
+## Call a Function:
+```javascript
+// Calling a function w/out arguments
+myCoolFunction();
+
+// Calling a function with arguments
+myCoolFunction(arg1, arg2);
+```
+
+When this code is read, `myCoolFunction()` is "called", all three statements within the function's body (those curly braces) get run.
+
+### Your Turn
+Let's go ahead and declare myCoolFunction in the console together and then call it. What do you notice about the code as it is being run? What do you notice about the code being run if you swap the 2nd and 3rd statements?
+
+Create your own functions in the console:
+
+- Write a function that declares a `firstName` variable and a `lastName` variable, then alerts the user with a message that incorporates the full name, and then logs a random number to the console.
+
+- Write a function that assigns three different math equations to three different variables, then log the sum of the values of all three variables
+
+- Write a function that alerts the user with a message of "YO!", and then logs to the console a sum of eight different integers.
+
+## Pass Information to a Function:
+Sometimes you need to give a function some information in order for it to do its job. You can give that function the information it needs by providing _parameters_. These are bits of information that you identify with appropriately named labels (you get to decide the notation) inside the parentheses of your named function. The words you use for your parameters act like variables INSIDE the function, which means they serve as means to pass values.
+
+### Parameters vs. Arguments
+Its a subtle difference. Basically, when you declare a function, and you stipulate the function will accept some bits of information, those are parameters. Then, when you pass the values of the parameters, those are called arguments. Like this:
+
+```javascript
+// parameters named on declaration of function
+function myDreamCar(make, model) {
+  return "Buy me " + make + " " + model;
+}
+
+const myDreamCar = (make, model) => {
```

Comment:
This lesson feels pretty large, especially because the goal is a very surface understanding. Wonder if it would make sense to cut all of the arrow function syntax for functions?

Reply:
Yes, you're right. We can probably cut this out without any impact to students' understanding or to the DOM lesson that follows.